syntax = "proto3";

service ChatService {
  rpc sendMessage (MessageRequest) returns (StatusResponse);
  rpc getMessages (MessageListRequest) returns (MessageListResponse);
}

message MessageRequest {
  string nickname = 1;
  string message = 2;
}

message StatusResponse {
  int32 status = 1; // >=0 for success (id assigned to the sent message), -1 for failure
  string status_message = 2;
}

message MessageListResponse {
  repeated string messages = 1; // list of messages
  int32 last_id = 2; // last message id sent
}

message MessageListRequest {
  int32 last_id = 1; // last message id received
  string nickname = 2;
}
